carp.deployment.core/src/commonMain/kotlin/dk/cachet/carp/deployment/domain/StudyDeploymentStatus.kt

```diff
@@ -19,17 +19,17 @@ data class StudyDeploymentStatus(
 )
 {
     /**
-     * Returns all [DeviceDescriptor]'s in [devicesStatus] which require registration.
+     * Returns all [AnyDeviceDescriptor]'s in [devicesStatus] which require registration.
      */
     fun getRemainingDevicesToRegister(): Set<AnyDeviceDescriptor> =
-        devicesStatus.filter { it.requiresRegistration && !it.isRegistered }.map { it.device }.toSet()
+        devicesStatus.filter { it.requiresRegistration && it !is DeviceDeploymentStatus.Registered }.map { it.device }.toSet()
```

Comment:
What if `it` is `Deployed` and somehow its `requiresRegistration` is set to `true`?

Reply:
You found a bug. ;p